dbt/scripts/export_qc_town_close_reports.py

```diff
@@ -0,0 +1,228 @@
+        help=(
+            "Print a command that can be run on the server to refresh "
+            "underlying iasWorld tables and exit. Useful if you want to "
+            "refresh table data before running exports"
+        ),
+    )
+    return parser.parse_args()
+
+
+def main():
+    """Main entrypoint for the script"""
+    args = parse_args()
+    tag = "tag:qc_report_town_close"
+    tag_suffix = "tri" if is_tri(args.township, args.year) else "non_tri"
+
+    if args.print_table_refresh_command:
+        # Use `dbt list` on parents to calculate update command for
+        # iasworld sources that are implicated by this call
+        dbt_list_args = [
+            "--quiet",
+            "list",
+            "--target",
+            args.target,
+            "--resource-types",
+            "source",
+            "--output",
+            "json",
+            "--output-keys",
+            "name",
+            "source_name",
+            "--select",
+            f"+{tag}",
+            f"+{tag}_{tag_suffix}",
+        ]
+        dbt_output = io.StringIO()
+        with contextlib.redirect_stdout(dbt_output):
+            dbt_list_result = DBT.invoke(dbt_list_args)
+
+        if not dbt_list_result.success:
+            print("Encountered error in `dbt list` call")
+            raise ValueError(dbt_list_result.exception)
+
+        # Output is formatted as a list of newline-separated JSON objects
+        source_deps = [
+            json.loads(source_dict_str)
+            for source_dict_str in dbt_output.getvalue().split("\n")
+            # Filter out empty strings caused by trailing newlines
+            if source_dict_str
+        ]
+        # Generate a Spark job definition for each iasWorld table that needs
+        # to be updated. For more context on what these attributes mean, see
+        # https://github.com/ccao-data/service-spark-iasworld
+        iasworld_deps = {}
+        for dep in source_deps:
+            table_name = dep["name"]
+            formatted_dep = {
+                "table_name": f"iasworld.{table_name}",
+                "min_year": args.year,  # Restrict to current year of data
+                "cur": ["Y"],  # Only get active records
+            }
+            # We should figure out a way to programmatically determine the
+            # correct filters for each table, but in the meantime, hardcode
+            # them based on known exceptions
+            if table_name == "sales":
+                # The sales table doesn't have a taxyr column
+                formatted_dep["min_year"] = formatted_dep["max_year"] = None
+            if table_name == "asmt_hist":
+                # The asmt_hist table doesn't have any cur = 'Y' records
+                del formatted_dep["cur"]
+
+            iasworld_deps[table_name] = formatted_dep
+
+        print("ssh into the server and run the following commands:")
+        print()
+        print("cd /home/shiny-server/services/service-spark-iasworld")
```

Comment:
Are you fine with storing this path under version control?

Reply:
Yep!